service/service.go

```diff
@@ -104,6 +126,12 @@ func NewService(cfg *Config) (*Service, error) {
 					log.Error(err)
 				}
 
+				if msg, err := serializeProofMsg(r); err != nil {
+					log.Error(err)
+				} else if err := broadcaster.BroadcastProof(msg); err != nil {
```

Comment:
consider calling `BroadcastProof` in a go routine, no reason to wait for the result (if you decide to change it, remember to add a log print in case of error inside the method, instead of here)